atomic_reactor/inner.py

```diff
@@ -538,7 +540,63 @@ def save(self, context_dir: ContextDir) -> None:
 
         logger.info("Writing workflow data into %s", context_dir.workflow_json)
         with open(context_dir.workflow_json, "w+") as f:
-            json.dump(self.dump(), f)
+            json.dump(self.as_dict(), f, cls=WorkflowDataEncoder)
+
+
+class WorkflowDataEncoder(json.JSONEncoder):
+    """Convert custom serializable objects into dict as JSON data."""
+
+    def default(self, o: object) -> Any:
+        if isinstance(o, ISerializer):
+            data = o.as_dict()
+            # Data type name used to know which type of object to recover.
+            data["__type__"] = o.__class__.__name__
+            return data
+        elif isinstance(o, ImageName):
+            return {
+                "__type__": o.__class__.__name__,
+                "str": o.to_str(),
+            }
+        elif isinstance(o, bytes):
+            return {
+                "__type__": "bytes",
+                "value": base64.b64encode(o).decode(),
+            }
```

Comment:
Instead of supporting bytes, could we change the the group_manifest plugin to return a regular string instead? I think the the `bytes` part of the result is only the 'manifest' key here https://github.com/containerbuildsystem/atomic-reactor/blob/e495e668e658b4f05bacc0f2c9c7ee57613a2e48/atomic_reactor/plugins/post_group_manifests.py#L137

and it should be OK to convert to a string